### PR TITLE
METAL-1039: Allow baremetal platform without MAPI

### PR DESCRIPTION
**Pull request by eranco74**
There are use cases for using the baremetal platform while disabling MAPI (for example to use CAPI).
The installer will skip rendering BMH, Machine and MachineSet CRs in case MAPI is disabled
This also removes the validation preventing this configuration to allow users to `create ignition-configs`.
The validations still apply for the `create cluster` target
    
Fixes [METAL-1039](https://issues.redhat.com//browse/METAL-1039)

### CODE REVIEW

File: pkg/asset/agent/installconfig_test.go [hunk 1]
```diff
@@ -4,11 +4,12 @@ import (
 	"net"
 	"testing"
 
-	"github.com/golang/mock/gomock"
-	"github.com/stretchr/testify/assert"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/pointer"
 
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
```
Comment (patrickdillon) [inline pkg/asset/agent/installconfig_test.go:11]: Judging by the multiple changes to imports like this, I think your formatting tools are configured in a way that's not going to be compatible with the installer linter. The linter will want all of these external packages to be grouped in a single block.

**Comment by eranco74** [comment]
@zaneb @patrickdillon please review

File: pkg/types/baremetal/validation/platform.go [hunk 2]
```diff
@@ -462,6 +457,48 @@ func ValidatePlatform(p *baremetal.Platform, agentBasedInstallation bool, n *typ
 	return allErrs
 }
 
+// GetEnabledCapabilities returns a set of enabled ClusterVersionCapabilities.
+func GetEnabledCapabilities(capabilities *types.Capabilities) sets.Set[configv1.ClusterVersionCapability] {
```
Comment (r4f4) [inline pkg/types/baremetal/validation/platform.go:461]: This is not specific to baremetal right? For the semantics of it, I'd prefer if it was in a more general place to be used by all platforms.

Comment (r4f4) [reply pkg/types/baremetal/validation/platform.go:461]: What about moving this to `pkg/types/installconfig.go`?
```
func (c *InstallConfig) EnabledCapabilities() sets.Set[configv1.ClusterVersionCapability] {
    [...]
}

**Review by eliorerz — APPROVED** [review]
/lgtm

**Comment by eranco74** [comment]
Unsure why this label is still required
/label acknowledge-critical-fixes-only

**Pull request merged**